src/resolve-config.js

```diff
@@ -19,23 +21,43 @@ function getLoadFunction(opts) {
   return getExplorerMemoized(opts).load;
 }
 
-function resolveConfig(filePath, opts) {
+function _resolveConfig(filePath, opts, sync) {
   opts = Object.assign({ useCache: true }, opts);
-  const load = getLoadFunction({ cache: !!opts.useCache, sync: false });
-  return load(filePath, opts.config).then(result => {
-    return !result ? null : mergeOverrides(result.config, filePath);
-  });
+  const loadOpts = { cache: !!opts.useCache, sync: !!sync };
+  const load = getLoadFunction(loadOpts);
+  const loadEditorConfig = resolveEditorConfig.getLoadFunction(loadOpts);
+  const arr = [load, loadEditorConfig].map(l => l(filePath, opts.config));
+
+  const unwrapAndMerge = arr => {
+    const result = arr[0];
+    const editorConfigged = arr[1];
+    return mergeEditorConfig(filePath, result, editorConfigged);
+  };
+
+  if (loadOpts.sync) {
+    return unwrapAndMerge(arr);
+  }
+
+  return Promise.all(arr).then(unwrapAndMerge);
 }
 
-resolveConfig.sync = (filePath, opts) => {
-  opts = Object.assign({ useCache: true }, opts);
-  const load = getLoadFunction({ cache: !!opts.useCache, sync: true });
-  const result = load(filePath, opts.config);
-  return !result ? null : mergeOverrides(result.config, filePath);
-};
+const resolveConfig = (filePath, opts) => _resolveConfig(filePath, opts, false);
+
+resolveConfig.sync = (filePath, opts) => _resolveConfig(filePath, opts, true);
+
+function mergeEditorConfig(filePath, result, editorConfigged) {
+  if (!filePath) {
```

Comment:
`null` should represent the config file not being found, so I guess **both** editor config and cosmiconfig failed.

Reply:
Yeah, I think I had to add that branch in to keep the pre-existing tests passing. (related to https://github.com/prettier/prettier/issues/2997)